cpp/src/binaryop/binaryop.cpp

```diff
@@ -33,6 +33,22 @@
 #include <string>
 #include <timestamps.hpp.jit>
 #include <types.hpp.jit>
+#include <libcudacxx/details/__config.jit>
+#include <libcudacxx/simt/cfloat.jit>
+#include <libcudacxx/simt/chrono.jit>
+#include <libcudacxx/simt/ctime.jit>
+#include <libcudacxx/simt/limits.jit>
+#include <libcudacxx/simt/ratio.jit>
+#include <libcudacxx/simt/type_traits.jit>
+#include <libcudacxx/simt/version.jit>
+#include <libcudacxx/libcxx/include/__config.jit>
+#include <libcudacxx/libcxx/include/__undef_macros.jit>
+#include <libcudacxx/libcxx/include/cfloat.jit>
+#include <libcudacxx/libcxx/include/chrono.jit>
+#include <libcudacxx/libcxx/include/ctime.jit>
+#include <libcudacxx/libcxx/include/limits.jit>
+#include <libcudacxx/libcxx/include/ratio.jit>
+#include <libcudacxx/libcxx/include/type_traits.jit>
```

Comment:
I'm in favor of this PR, but man this is painful. Everything I can think to do to automate this seems like more work than manually hunting down the compilation errors :/.

Reply:
If we don't plan on having conditional includes in JIT kernels, I think we could automate it with some effort.  It's definitely more work than putting a print statement in the current header handler and adding any missing headers, but the effort could be worthwhile if we're expecting to update the list often.

At build time, we could build and invoke a program that calls jitify on each kernel, using a special only-at-build-time header callback handler that:
* records the mapping of the requested path to the header file
* loads the header file as a string and returns it, so jitify can proceed to find subsequent headers

The recorded mapping of requested paths to headers can be used to both build the hash table needed by the runtime header handler and the list of header files to stringify.  Then we can stringify the necessary headers and generate a file that contains the `#include` directives for the stringified headers along with the hash table that maps requested paths to those stringified headers.  That generated file can serve as a header handler for the various JIT kernel use-cases like in binaryop.cpp.

I didn't go to that extent initially since I wanted to get _something_ working to fix the issues we were seeing.

Reply:
Thanks, that description sounds like our best bet. I might take a stab at implementing a dev utility for doing this automatically if it gets to be a headache for people.